apollo-router/src/query_planner/fetch.rs

```diff
@@ -168,21 +169,20 @@ impl SubgraphOperation {
     pub(crate) fn as_parsed(
         &self,
         subgraph_schema: &Valid<apollo_compiler::Schema>,
-    ) -> &Arc<Valid<ExecutableDocument>> {
-        self.parsed.get_or_init(|| {
+    ) -> Result<&Arc<Valid<ExecutableDocument>>, ValidationErrors> {
+        self.parsed.get_or_try_init(|| {
```

Comment:
Yes, but after the first such call the error would be propagated up the call stack and turned into a GraphQL error for the original client, so I expect there wouldn’t be a second call.

Reply:
I am not too happy about that `OnceLock`. In `hash_subqueries` we work on `&mut self` and update the field once and for all. Using `as_parsed` and a `OnceLock` is hacking around the issue

Reply:
When I first wrote this apollo-federation was using `Arc` in its own query plan data structure and the intent was to migrate Router to use that one eventually. So I opted not to count of `&mut` being easily available. But it looks like these `Arc` has been removed there since, for fetch nodes.

We can make `as_serialized` eagerly-initialized, but since subgraphs schemas are not available during `serde::Deserialize` then `as_parsed` still needs to be lazily-initialized. Do you think doing so with `&mut Option` is an improvement over `&OnceLock`?